build-sonarlint/common.mjs

```diff
@@ -0,0 +1,71 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2022 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+import { createVSIX } from 'vsce';
+import { clean } from './fsUtils.mjs';
+import { info } from 'fancy-log';
+import updateVersion from './updateVersion.mjs';
+import downloadJre from './jreDownload.mjs';
+import cycloneDx from './sbomGeneration.mjs';
+import { computeUniversalVsixHashes } from './hashes.mjs';
+import { deployBuildInfo, deployVsix } from './deployUtils.mjs';
+import signVsix from './sign.mjs';
+import _default from './constants.mjs';
+const { TARGETED_PLATFORMS, LATEST_JRE } = _default;
+
+export async function deployUniversal() {
+  commonPreBuildTasks();
+  await createVSIX();
+  await commonPostBuildTasks();
+}
+
+export async function deployAll() {
+  commonPreBuildTasks();
+  await createVSIX();
+  await buildTargeted();
+  await commonPostBuildTasks();
+}
+
+async function buildTargeted() {
+  for (const platform of TARGETED_PLATFORMS) {
+    await executeWithDurationLog(async () => {
+      await buildForPlatform(platform);
+    }, `Build-for-${platform}-platform`);
+  }
+}
+
+async function buildForPlatform(platform) {
+  await downloadJre(platform, LATEST_JRE);
+  await createVSIX({ target: platform });
+}
+
+function commonPreBuildTasks() {
+  clean();
+  updateVersion();
+  cycloneDx();
+}
+
+async function commonPostBuildTasks() {
+  computeUniversalVsixHashes();
+  await signVsix({
+    privateKeyArmored: process.env.GPG_SIGNING_KEY,
+    passphrase: process.env.GPG_SIGNING_PASSPHRASE
+  });
+  await deployBuildInfo();
+  await executeWithDurationLog(async () => {
+    await deployVsix();
+  }, 'Deploy-vsix');
+}
+
+export async function executeWithDurationLog(callback, taskName) {
```

Comment:
Maybe wrap each deployment stage into this logging helper.

Reply:
We had it that way and removed it later, because it makes code much less readable, without too much additional value. So we only kept them for more "long" and "heavy" tasks